src/js/components/StatusIcon.js

```diff
@@ -0,0 +1,56 @@
+import React, { PropTypes, Component } from "react";
+
+import Icon from "#SRC/js/components/Icon";
+
+const ICONS_STATE = {
+  RUNNING: {
+    id: "circle-check",
+    color: "green"
+  },
+  TRANSITION: {
+    id: "spinner",
+    color: "grey"
+  },
+  STOPPED: {
+    id: "circle-minus",
+    color: "light-grey"
+  },
+  WARNING: {
+    id: "yield",
+    color: "yellow"
+  }
+};
+
+/**
+ * Status Icon is meant to be used across
+ * the application where it need to display any of the state below
+ * All green (RUNNING), transition (DEPLOYING, RECOVERING), Warning
+ *
+ * @class StatusIcon
+ * @extends {Component}
+ */
+class StatusIcon extends Component {
+  render() {
+    const state = this.props.state;
+    const iconState = ICONS_STATE[state];
+
+    if (iconState == null) {
+      return null;
+    }
+
+    return (
+      <Icon
+        className={iconState.id}
+        color={iconState.color}
+        id={iconState.id}
+        size="mini"
+      />
+    );
+  }
+}
+
+StatusIcon.propTypes = {
+  state: PropTypes.oneOfType(["RUNNING", "TRANSITION", "STOPPED", "WARNING"])
```

Comment:
Should be [oneOf](https://reactjs.org/docs/typechecking-with-proptypes.html) this is throwing an error in the console right now.

Reply:
thanks

Reply:
As @Poltergeist suggested below, what about `statusClassname`?